airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/FeatureFlagFetchActivityImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import io.airbyte.api.client.AirbyteApiClient;
+import io.airbyte.api.client.generated.WorkspaceApi;
+import io.airbyte.api.client.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.generated.WorkspaceRead;
+import io.airbyte.featureflag.FeatureFlagClient;
+import io.airbyte.featureflag.FieldSelectionEnabled;
+import io.airbyte.featureflag.Flag;
+import io.airbyte.featureflag.Workspace;
+import jakarta.inject.Singleton;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class FeatureFlagFetchActivityImpl implements FeatureFlagFetchActivity {
+
+  private final WorkspaceApi workspaceApi;
+  private final FeatureFlagClient featureFlagClient;
+
+  public FeatureFlagFetchActivityImpl(WorkspaceApi workspaceApi,
+                                      FeatureFlagClient featureFlagClient) {
+    this.workspaceApi = workspaceApi;
+    this.featureFlagClient = featureFlagClient;
+  }
+
+  public UUID getWorkspaceId(final UUID connectionId) {
+    final WorkspaceRead workspace = AirbyteApiClient.retryWithJitter(
```

Comment:
Nit: I don't think than the retry with Jitter is needed. The activity option (shortRunningActivity) has its own retry with a backoff coefficient. Having 2 retries seem unnecessary.